controllers/configmap_controller.go

```diff
@@ -168,6 +168,29 @@ func validateAddress(address string) error {
 	return nil
 }
 
+// getNodeUsername retrieves the username associated with the given node from the instance ConfigMap data
+func getNodeUsername(configMapData map[string]string, node *core.Node) (string, error) {
+	if node == nil {
+		return "", errors.New("cannot get username for nil node")
+	}
+	// Find entry in ConfigMap that is associated to node via address
+	for _, address := range node.Status.Addresses {
+		if data, found := configMapData[address.Address]; found {
+			return extractUsernameFromEntry(data)
+		}
+	}
+	return "", errors.Errorf("unable to find instance associated with node %s", node.GetName())
+}
+
+// extractUsernameFromEntry returns the username string from data in the form username=<username>
+func extractUsernameFromEntry(value string) (string, error) {
+	splitData := strings.SplitN(value, "=", 2)
+	if len(splitData) == 0 || splitData[0] != "username" {
+		return "", errors.New("data has an incorrect format")
```

Comment:
consider logging the incorrectly formatted data (value)

Reply:
That's a valid concern; But, whoever has access to the logs, is authenticated already

Reply:
Please don't log the username to be on the safe side

Reply:
We are logging the username when initializing the connection to the instance. Not sure if we should be removing that too now.

Reply:
I think it is a good idea to remove, given the work we are doing to protect the plaintext value. I'll add a separate commit removing from that log.

Reply:
I can take care of it in my logging PR. Please remind me during the PR review.

Reply:
On second thought that belongs in this PR.